src/hm-search.js

```diff
@@ -191,17 +191,33 @@ function scoreSort(x) {
   return x.sort(scoreCmp);
 }
 
-function nameSearch(db, inputName) {
-  if (inputName === '*') {
+function nameSearch(db, input, opts) {
+  if (opts.fuzzy) {
+    return nameSearchFuzzy(db, input);
+  }
+  else {
+    return nameSearchSubstring(db, input);
+  }
+}
```

Comment:
In my view this function should not exist, as it's really *two* functions. I suggest moving the conditional logic to the call site:

```javascript
opts.fuzzy ? nameSearchFuzzy(db, input) : nameSearchSubstring(db, input)
```